src/profile-logic/call-node-samples.js

```diff
@@ -0,0 +1,221 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import type { Thread } from '../types/profile';
+import type { Milliseconds } from '../types/units';
+import type {
+  IndexIntoCallNodeTable,
+  CallNodeTable,
+  CallNodeInfo,
+} from '../types/profile-derived';
+
+import { getCallNodePath } from './profile-data';
+
+export type CallNodeSamplesByDepth = Array<{
+  start: Milliseconds[],
+  end: Milliseconds[],
```

Comment:
The x-axis of a flame graph does not represent time. So these units should probably change, and can perhaps instead be represented as some kind of width. But I don't want to make too many changes before I know that I am on the right track.